atlasaction/action_test.go

```diff
@@ -87,6 +91,200 @@ func TestMigrateApply(t *testing.T) {
+		tt := newT(t)
+		tt.setupConfigWithLogin(t, srv.URL, token)
+		tt.setInput("dir", "file://testdata/migrations")
+		tt.setInput("dir-name", "test-#dir")
+		tt.setInput("dev-url", "sqlite://file?mode=memory")
+		err := MigratePush(context.Background(), tt.cli, tt.act)
+		require.ErrorContains(t, err, "slug must be lowercase alphanumeric")
+	})
+}
+
+func TestMigrateE2E(t *testing.T) {
+	type (
+		pushDir struct {
+			Slug   string `json:"slug"`
+			Tag    string `json:"tag"`
+			Driver string `json:"driver"`
+			Dir    string `json:"dir"`
+		}
+		syncDir struct {
+			Slug    string        `json:"slug"`
+			Driver  string        `json:"driver"`
+			Dir     string        `json:"dir"`
+			Context *ContextInput `json:"context"`
+		}
+
+		graphQLQuery struct {
+			Query     string          `json:"query"`
+			Variables json.RawMessage `json:"variables"`
+			PushDir   struct {
+				pushDir `json:"input"`
+			}
+			SyncDir struct {
+				syncDir `json:"input"`
+			}
+		}
+	)
+
+	var syncDirCalls, pushDirCalls int
+	token := "123456789"
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		require.Equal(t, "Bearer "+token, r.Header.Get("Authorization"))
+		query := graphQLQuery{}
+		err := json.NewDecoder(r.Body).Decode(&query)
+		require.NoError(t, err)
+		switch {
+		case strings.Contains(query.Query, "syncDir"):
+			syncDirCalls++
+			require.NoError(t, json.Unmarshal(query.Variables, &query.SyncDir))
+			require.Equal(t, "test-dir", query.SyncDir.Slug)
+			expected := &ContextInput{
+				Repo:   "repository",
+				Path:   "file://testdata/migrations",
+				Branch: "testing-branch",
+				Commit: "sha1234",
+				URL:    "",
+			}
+			require.Equal(t, expected, query.SyncDir.Context)
+		case strings.Contains(query.Query, "pushDir"):
+			pushDirCalls++
+			require.NoError(t, json.Unmarshal(query.Variables, &query.PushDir))
+			require.Equal(t, query.PushDir.Tag, "sha1234")
+			require.Equal(t, query.PushDir.Slug, "test-dir")
+			fmt.Fprint(w, `{"data":{"pushDir":{"url":"https://some-org.atlasgo.cloud/dirs/314159/tags/12345"}}}`)
+		}
+	}))
+	t.Cleanup(srv.Close)
+
+	tt := newT(t)
+	tt.setupConfigWithLogin(t, srv.URL, token)
+	tt.setInput("dir", "file://testdata/migrations")
+	tt.setInput("dir-name", "test-dir")
+	tt.setInput("dev-url", "sqlite://file?mode=memory")
+	tt.env["GITHUB_REPOSITORY"] = "repository"
+	tt.env["GITHUB_REF_NAME"] = "testing-branch"
+	tt.env["GITHUB_SHA"] = "sha1234"
+	err := MigratePush(context.Background(), tt.cli, tt.act)
+	require.NoError(t, err)
+	require.Equal(t, syncDirCalls, 1)
+	require.Equal(t, pushDirCalls, 1)
+	require.NoError(t, err)
+	outputs, _ := tt.outputs()
+	url := outputs["url"]
+	require.Equal(t, "https://some-org.atlasgo.cloud/dirs/314159/tags/12345", url)
+}
+
+func generateHCL(t *testing.T, url, token string) string {
+	tmpl := `
+	atlas {
+		cloud {
+			token = "{{ .Token }}"
+		{{- if .URL }}
+			url = "{{ .URL }}"
+		{{- end }}
+		}	  
+	}
+	env "test" {
+  	}
```

Comment:
```suggestion
	env {
	  name = atlas.env
  	}
```
See "unlabeled" blocks 
https://atlasgo.io/atlas-schema/projects#flags

Reply:
This was added to remove flexibility, we have tests that are using broken env names on purpose